tests/unit_node/vm_test.ts

```diff
@@ -10,16 +17,76 @@ Deno.test({
   },
 });
 
+Deno.test({
+  name: "vm new Script()",
+  fn() {
+    const script = new Script(`
+function add(a, b) {
+  return a + b;
+}
+const x = add(1, 2);
+x
+`);
+
+    const value = script.runInThisContext();
+    assertEquals(value, 3);
+  },
+});
+
+// https://github.com/denoland/deno/issues/23186
 Deno.test({
   name: "vm runInNewContext sandbox",
   fn() {
-    assertThrows(() => runInNewContext("Deno"));
-    // deno-lint-ignore no-var
-    var a = 1;
-    assertThrows(() => runInNewContext("a + 1"));
+    const sandbox = { fromAnotherRealm: false };
+    runInNewContext("fromAnotherRealm = {}", sandbox);
+
+    assertEquals(typeof sandbox.fromAnotherRealm, "object");
+  },
+});
+
+// https://github.com/denoland/deno/issues/22395
+Deno.test({
+  name: "vm runInewContext with context object",
+  fn() {
+    const context = { a: 1, b: 2 };
+    const result = runInNewContext("a + b", context);
+    assertEquals(result, 3);
+  },
+});
+
+// https://github.com/denoland/deno/issues/18299
+Deno.test({
+  name: "vm createContext and runInContext",
+  fn() {
+    // @ts-expect-error implicit any
+    globalThis.globalVar = 3;
+
+    const context = { globalVar: 1 };
+    createContext(context);
+    runInContext("globalVar *= 2", context);
```

Comment:
I don't understand how it works if `context` argument is ignored in `ContextifyScript::eval_machine`

Reply:
https://github.com/denoland/deno/pull/23202#discussion_r1555164971